Include/pystate.h

```diff
@@ -48,6 +48,9 @@ typedef struct _is {
     PyObject *import_func;
     /* Initialized to PyEval_EvalFrameDefault(). */
     _PyFrameEvalFunction eval_frame;
+
+    Py_ssize_t co_extra_user_count;
+    freefunc co_extra_freefuncs[MAX_CO_EXTRA_USERS];
 } PyInterpreterState;
```

Comment:
One possibility here would be maintaining a separate list for the code extra states.  So:

struct PyCodeExtraState {
    PyCodeExtraState* next;
    PyInterpreterState* interp;
}

And then update the list when interpreter states are created/destroyed, and search the list for the current interprerter when getting the state.  Then leave the thread state fields around with the preserve ABI names.

Does that sound better then risking changing PyInterpreterState?